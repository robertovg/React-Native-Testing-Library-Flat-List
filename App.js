import React, { Component } from 'react';
import { Text, View, StyleSheet, FlatList } from 'react-native';
import { projectsMock } from './data/projectsMock';
import AppWithFlatList from './AppWithFlatList';
import AppWithoutFlatList from './AppWithoutFlatList';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      projects: [...projectsMock],
    };
  }

  render() {
    const { projects } = this.state;
    // return <AppWithoutFlatList projects={projects} />;
    return <AppWithFlatList projects={projects} />;
  }
}
