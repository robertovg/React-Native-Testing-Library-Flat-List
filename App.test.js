import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import App from './App';

describe('<App', () => {
  it('Simple snapshot of the App, now loading FlatList, and not render', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});
