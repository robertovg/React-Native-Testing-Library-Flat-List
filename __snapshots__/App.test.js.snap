// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App Simple snapshot of the App, now loading FlatList, and not render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "justifyContent": "center",
      "padding": 8,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "margin": 24,
        "textAlign": "center",
      }
    }
    testID="header"
  >
    After this text, we should have the FlatList with the Projects
  </Text>
</View>
`;
